### ISSUE TITLE
Comments

### ISSUE THREAD

**Issue by bucko13** [bpanel-org/price#1]
- fiat symbols: I wonder if there's an API we can hit for that
- feeds: Instead of using `https` it'd probably be better to use `bcurl`. It's got async/await support and also parses the responses for you so it's a lot easier to manage. I'm actually surprised that worked since `https` is a nodejs module. In the browser you would usually use `fetch`. My guess is it was a polyfill by webpack or babel (another reason to use something that's going to work "normally" in the browser)
- feeds: I feel like these could actually just be a store of strings of the endpoints to use as a part of the redux store. Then you have middleware that essentially retrieves the currently set price feed endpoint and calls whatever that is set to. The advantage of this construction is that developers can add whatever feeds they want without even needing to submit a PR since you would just need to decorate your store.

I think that's it :). Just style comments basically. It works and it follows the important conventions which are the most important things!

**Comment by bucko13** [comment]
One more thing- we might want to think of changing the API around `UPDATE_AVAIL_FIATS`. Right now, the reducer just overwrites whatever is in the payload. I could see this causing a problem though if the feed returns a format that's unexpected. It also means another plugin could potentially mess with this by just returning some arbitrary payload. This might be able to be streamlined a bit if the feed info is stored in the store as well and the feed fetching is built around that.

**Comment by pinheadmz** [comment]
You're right on about `https` and `bcurl` -- not sure why I didn't look at b suite first for everything! Will update.

About `UPDATE_AVAIL_FIATS` -- the fiat list needs to be in the store so plugins like price-widget can select it. Is there any way to update something in the store _without_ it also being available to every other plugin? Like, is the same thing true about client, transaction, and block data? Any plugin can just dispatch arbitrary data to the store? The way it's structured now, I have a function for EACH API that returns an array of available fiats. I suppose I could add better error-checking to this, but the format should always be an array of 3-letter acronyms.

Similar response to letting other plugins add new APIs - APIs all have different response shapes, so it wouldn't necessarily just be a string that represents an endpoint, it'd also have to include a function to parse the returned JSON data.

For example the coinbase API returns:
```
{"trade_id":55468857,"price":"3621.00000000","size":"0.02665765","bid":"3620.99","ask":"3621","volume":"25395.29287570","time":"2018-12-06T20:39:35.976000Z"}
```

but the blockchain.info API returns:
```
{
  "USD" : {"15m" : 3667.26, "last" : 3667.26, "buy" : 3667.26, "sell" : 3667.26, "symbol" : "$"},
  "AUD" : {"15m" : 5075.46, "last" : 5075.46, "buy" : 5075.46, "sell" : 5075.46, "symbol" : "$"},
  "BRL" : {"15m" : 14222.75, "last" : 14222.75, "buy" : 14222.75, "sell" : 14222.75, "symbol" : "R$"},
  "CAD" : {"15m" : 4907.87, "last" : 4907.87, "buy" : 4907.87, "sell" : 4907.87, "symbol" : "$"},
  "CHF" : {"15m" : 3639.64, "last" : 3639.64, "buy" : 3639.64, "sell" : 3639.64, "symbol" : "CHF"},
  "CLP" : {"15m" : 2478383.73, "last" : 2478383.73, "buy" : 2478383.73, "sell" : 2478383.73, "symbol" : "$"},
  "CNY" : {"15m" : 25241.05, "last" : 25241.05, "buy" : 25241.05, "sell" : 25241.05, "symbol" : "¥"},
  "DKK" : {"15m" : 24042.26, "last" : 24042.26, "buy" : 24042.26, "sell" : 24042.26, "symbol" : "kr"},
  "EUR" : {"15m" : 3202.63, "last" : 3202.63, "buy" : 3202.63, "sell" : 3202.63, "symbol" : "€"},
  "GBP" : {"15m" : 2867.86, "last" : 2867.86, "buy" : 2867.86, "sell" : 2867.86, "symbol" : "£"},
  "HKD" : {"15m" : 28655.12, "last" : 28655.12, "buy" : 28655.12, "sell" : 28655.12, "symbol" : "$"},
  "INR" : {"15m" : 259083.04, "last" : 259083.04, "buy" : 259083.04, "sell" : 259083.04, "symbol" : "₹"},
  "ISK" : {"15m" : 445059.12, "last" : 445059.12, "buy" : 445059.12, "sell" : 445059.12, "symbol" : "kr"},
  "JPY" : {"15m" : 407971.53, "last" : 407971.53, "buy" : 407971.53, "sell" : 407971.53, "symbol" : "¥"},
  "KRW" : {"15m" : 4106712.39, "last" : 4106712.39, "buy" : 4106712.39, "sell" : 4106712.39, "symbol" : "₩"},
  "NZD" : {"15m" : 5332.96, "last" : 5332.96, "buy" : 5332.96, "sell" : 5332.96, "symbol" : "$"},
  "PLN" : {"15m" : 13785.05, "last" : 13785.05, "buy" : 13785.05, "sell" : 13785.05, "symbol" : "zł"},
  "RUB" : {"15m" : 245063.83, "last" : 245063.83, "buy" : 245063.83, "sell" : 245063.83, "symbol" : "RUB"},
  "SEK" : {"15m" : 32979.88, "last" : 32979.88, "buy" : 32979.88, "sell" : 32979.88, "symbol" : "kr"},
  "SGD" : {"15m" : 5023.33, "last" : 5023.33, "buy" : 5023.33, "sell" : 5023.33, "symbol" : "$"},
  "THB" : {"15m" : 120573.53, "last" : 120573.53, "buy" : 120573.53, "sell" : 120573.53, "symbol" : "฿"},
  "TWD" : {"15m" : 113234.27, "last" : 113234.27, "buy" : 113234.27, "sell" : 113234.27, "symbol" : "NT$"}
}
```

**Comment by bucko13** [comment]
For the `UPDATE_AVAIL_FIATS`, I think it could still be in the store, but it might be good to add additional checks on the payload. Just confirm it's an array, or add as an individual item with each dispatch and confirm the item is a string. That way you're at least not just pushing whatever is passed onto the store.

The feeds is definitely tricky so we'd have to think about how that would be done. Might be possible to do parsing of the returns in a middleware. Then a plugin that has added a new feed would also have a middleware that parses data it's going to add. But yeah, this definitely isn't as important. Mostly just spitballing about how that could be formatted with some additional flexibility in the future.
